server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -52,19 +62,25 @@ private static final class VersionLookup {
         // the tombstone
         private final AtomicLong minDeleteTimestamp = new AtomicLong(Long.MAX_VALUE);
 
+        public static VersionLookup merge(VersionLookup v1, VersionLookup v2) {
+            var newMap = v1.map;
+            newMap.putAll(v2.map);
+            return new VersionLookup(newMap);
```

Comment:
I am not sure why we need a new instance when we modify the original map anyway?